src/main/java/com/jlr/ttl/ds/api/JwtAuthorizationFilter.java

```diff
@@ -0,0 +1,104 @@
+package com.jlr.ttl.ds.api;
+
+import com.jlr.ttl.ds.api.constants.DSConstants;
+import com.jlr.ttl.ds.api.dto.DSResponse;
+import com.jlr.ttl.ds.api.dto.authentication.JwtUtils;
+import com.jlr.ttl.ds.api.dto.response.VehicleResponse;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+@Configuration
+public class JwtAuthorizationFilter extends OncePerRequestFilter {
+    JwtUtils token;
+    List<String> authoritiesStringList = new ArrayList<>();
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+
+        String jwt = getJwtFromRequest(request);
+        authoritiesStringList.clear();
+        if (jwt != null) {
+            try {
+                Claims claims = Jwts.parser().setSigningKey("secret").parseClaimsJws(jwt).getBody();
+                List<GrantedAuthority> authorities = extractPermissions(claims);
+                Authentication auth = new UsernamePasswordAuthenticationToken(null, null,authorities);
+
+                if(request.getServletPath().startsWith("/v1/vehicles/all")){
+                    if(authoritiesStringList.contains("Vehicles")){
+                        response.setStatus(HttpStatus.OK.value());
+                    }else{
+                        auth.setAuthenticated(false);
+                    }
+                }else if(request.getServletPath().startsWith("/v1/vehicles/by_id/")) {
+                    if (authoritiesStringList.contains("Admin")) {
```

Comment:
When you change this to a switch case, instead of an if else use a simple if and break if a condition is true

Reply:
This has been resolved